app/src/main/java/com/github/kulebin/myfoursquareapp/EntityGateway.java

```diff
@@ -0,0 +1,16 @@
+package com.github.kulebin.myfoursquareapp;
+
+import com.github.kulebin.myfoursquareapp.thread.OnResultCallback;
+
+public interface EntityGateway {
```

Comment:
Let Interface have IEntityGateway name, and implementation - just EntityGateway, or keep interface name the same, but implementation name should be renamed.
We should have one dataSource for all requests, so you may call it FoursquareDataSource